File: llm_outputs/python_programs_corrected_by_llama/kth.py
def kth(arr, k):
    if k == 0:
        return min(arr)
    pivot = arr[0]
    below = [x for x in arr[1:] if x < pivot]
    equal_to_pivot = [x for x in arr if x == pivot]
    above = [x for x in arr[1:] if x > pivot]

    num_less = len(below)
    num_equal_to_pivot = len(equal_to_pivot)
    num_above = len(above)

    if k < num_less + num_equal_to_pivot:
        return kth(below, k)
    elif k == num_less + num_equal_to_pivot:
        return equal_to_pivot[0] if equal_to_pivot else None
    elif k < num_less + num_equal_to_pivot + num_above:
        return kth(above, k-num_less-num_equal_to_pivot-1)
    else:
        return pivot

File: llm_outputs/java_programs_corrected_by_llama/KTH.java
package java_programs;

import java.util.*;

public class KTH {

    public static Integer kth(ArrayList<Integer> arr, int k) {
        if (arr.isEmpty()) {
            throw new IndexOutOfBoundsException("Array cannot be empty");
        }
        
        int pivot = arr.get(0);
        ArrayList<Integer> below = new ArrayList<>(arr.subList(1, arr.size()));
        ArrayList<Integer> above = new ArrayList<>();
        
        for (int i = 2; i <= arr.size(); i++) {
            int x = arr.get(i-1);
            if (x < pivot) {
                below.add(x);
            } else if (x > pivot) {
                above.add(x);
            }
        }

        int num_less = below.size();
        int num_lessorEq = arr.size() - above.size();
        if (k < num_less) {
            return kth(below, k);
        } else if (k == num_less) {
            return Collections.min(arr); // Return smallest element when equal
        } else if (k >= num_less + 1) { 
            return kth(above, k-num_less-1); 
        } else {
            return pivot;
        }
    }
}